src/tools/yarprobotstatepublisher/CMakeLists.txt

```diff
@@ -1,17 +1,22 @@
-# Copyright (C) 2017 Fondazione Istituto Italiano di Tecnologia (IIT)  
+# Copyright (C) 2017 Fondazione Istituto Italiano di Tecnologia (IIT)
 # All Rights Reserved.
 # Authors: Silvio Traversaro <silvio.traversaro@iit.it>
 
 project(yarprobotstatepublisher)
 
-set(ROS_IDL_FILES JointState.msg)
-yarp_add_idl(IDL_GEN_FILES ${ROS_IDL_FILES})
-message("IDL_GEN_FILES : " ${IDL_GEN_FILES})
-add_executable(${PROJECT_NAME} include/robotstatepublisher.h src/main.cpp src/robotstatepublisher.cpp ${IDL_GEN_FILES})
-target_include_directories(${PROJECT_NAME} PRIVATE ${PROJECT_SOURCE_DIR}/include
-                                                     ${YARP_INCLUDE_DIRS})
-# Workaround for yarp_add_idl problems
-target_include_directories(${PROJECT_NAME} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
+find_package(YARP COMPONENTS rosmsg REQUIRED)
+
+add_executable(${PROJECT_NAME}
+    include/robotstatepublisher.h
+    src/main.cpp
+    src/robotstatepublisher.cpp)
+
+target_include_directories(${PROJECT_NAME} PRIVATE ${PROJECT_SOURCE_DIR}/include)
+
+target_link_libraries(${PROJECT_NAME}
+    ${YARP_LIBRARIES}
+    YARP::YARP_rosmsg
+    idyntree-high-level
+    idyntree-yarp)
```

Comment:
I am not following everything in this changes (a typical problem in big PRs), but I guess that now we can remove the file https://github.com/robotology/idyntree/blob/master/src/tools/yarprobotstatepublisher/JointState.msg .

Reply:
I agree on that. Done https://github.com/robotology/idyntree/pull/538/commits/bdb55caaa8689852faa5f172236b681cc42b31a7